content/blog/clickhouse-benchmarking.md

```diff
@@ -0,0 +1,188 @@
+
+In this blog, we’ll focus on the performance benchmarking of ClickHouse on AWS ECS during the ingestion of different data volumes. We’ll analyze key system metrics such as CPU usage, memory consumption, disk I/O, and row insertion rates across varying data ingestion sizes.
+
+For setting up the ClickHouse cluster, we followed the [ClickHouse replication architecture guide](https://clickhouse.com/docs/en/architecture/replication) and the [AWS CloudFormation ClickHouse cluster setup](https://aws-ia.github.io/cfn-ps-clickhouse-cluster/). Using these resources, we replicated the setup on ECS, allowing us to run performance benchmarking tests on the environment.
+
+By examining performance metrics during the ingestion of 1 million (10 lakh), 5 million (50 lakh), 10 million (1 crore), and 66 million (6.6 crore) logs, we aim to provide a quantitative analysis of how system behavior changes as the load increases.
+
+## Performance Comparison of Key Metrics Across Ingestion Volumes
+
+| **Logs Ingested** | **CPU Usage (Cores)** | **Selected Bytes per Second (B/s)** | **IO Wait (s)** | **CPU Wait (s)** | **Read from Disk (B)** | **Read from Filesystem (B)** | **Memory Tracked (Bytes)** | **Selected Rows per Second** | **Inserted Rows per Second** |
+|-------------------|-----------------------|------------------------------------|-----------------|-----------------|------------------------|------------------------------|----------------------------|------------------------------|------------------------------|
+| **1 Million (10 Lakh)** | 0.103 | 27,118.35 | 0.000 | 0.0217 | 0.000 | 439,023.78 | 714,778,747.45 | 256.88 | 238,666.67 |
+| **5 Million (50 Lakh)** | 0.111 | 37,545.82 | 0.000 | 0.0058 | 0.000 | 532,938.08 | 738,249,399.13 | 256.88 | 238,666.67 |
+| **10 Million (1 Crore)** | 0.433 | 36,265,260.35 | 0.000 | 0.3297 | 1,843.20 | 3,323,855.43 | 1,030,697,939.98 | 146,848.98 | 140,193.58 |
+| **66 Million (6.6 Crore)** | 1.179 | 122,710,173.00 | 0.750 | 4.4276 | 17,330,176.00 | 14,303,693.00 | 2,065,638,032.00 | 365,589.00 | 365,589.00 |
+
+<!-- markdownlint-disable MD033 -->
+<p align="center">
+  <img src="/images/blog/clickhouse-benchmarking/clickhouse-write-operations.png" alt="clickhouse-benchmarking" style="border-radius: 10px; width: 300; height: 500;">
+</p>
+<!-- markdownlint-enable MD033 -->
+
+### Key Insights from the Data
+
+#### 1. **CPU Usage (avg_cpu_usage_cores)**
+
+- **At 1 Million logs**, the CPU usage was minimal at **0.103 cores**, indicating a low load on the system.
```

Comment:
For values, you can use code blocks to highlight.

 `0.103 cores` over **0.103 cores**. I think bold is overly used in the blog from heading to data like **At 1 Million logs** and results **0.103 cores**.
 
 I think we can improve readability.